### PR TITLE
Classify and downgrade bufrw connection errors

### PR DESCRIPTION
**Pull request by jcorbin**
r @kriskowal @raynos

### CODE REVIEW

File: errors.js [hunk 1]
```diff
@@ -648,6 +648,10 @@ Errors.classify = function classify(err) {
         return err.codeName;
     }
 
+    if (err.type.indexOf('bufrw.') > -1) {
+        return 'NetworkError';
```
Comment (rf) [inline errors.js]: Not `ProtocolError`?

Comment (jcorbin) [inline errors.js]: Yeah, in particular what we're seeing is a bufrw short error in connection closing.  Seems like remotes side is closing mid-frame-write.

**Comment by jcorbin** [comment]
Updated based on bufrw 0.9.27 to clarify read vs write vs internal errors

**Comment by Raynos** [comment]
lgtm.

**Comment by kriskowal** [comment]
so say we all

**Pull request merged**